src/requests.js

```diff
@@ -64,6 +64,44 @@ export function updateRequest(path, { data, permissions }, options = {}) {
 /**
  * @private
  */
+export function jsonPatchPermissionsRequest(
+  path,
+  permissions,
+  opType,
+  options = {}
+) {
+  const {
+    headers,
+    safe,
+    last_modified,
+  } = { ...requestDefaults, ...options };
+
+  const ops = [];
+
+  for (const type in permissions) {
+    for (const principal in permissions[type]) {
```

Comment:
It's actually `const principal for permissions[type]` with arrays. It works because with `in` here we're using the indexes, but it's more semantically correct to use `for .. of`. Also you can then directly use `principal` instead of `permissions[type][principal]`.